### ISSUE TITLE
getHref should return the most specific route

### ISSUE THREAD

**Issue by pzpsofficial** [svobik7/next-roots#109]
Hi! 

First of all, thank you for great work, it really is a life saver ;) 

I have the following structure of my pages: 
```
[slug] // directory for pages that can be dynamically created via CMS
page.tsx // for home page
news // directory for news
```

When I call
```ts
const route = router.getRouteFromHref('/news');
```

I get:
```ts
{
    name: '/[slug]',
    href: '/locale/:slug'
}
```

Am I missing something? Or is it connected with one of opened issues?

**Comment by svobik7** [comment]
Hi @pzpsofficial thank you for creatin your PR 👍 It is generally good I have just added some of my ideas. Once you have a time please check my suggestions. Thank you

**Comment by pzpsofficial** [comment]
Hi @svobik7! I will take a look at it tomorrow :D

**Comment by svobik7** [comment]
Closing. Fixed in new 3.2.1 version. Thank you @pzpsofficial
